File: 0151-reverse-words-in-a-string/solution.cpp
class Solution {
public:
    string reverseWords(string s) {
        const int n = s.size();
        vector<string> v ;
        stringstream ss(s) ;
        string word ;
        while(ss >> word){
            v.push_back(word);
        }
        
        string res="";
        for(int i = v.size() - 1 ; i >= 0 ; i--) res+= v[i] + " ";
        res.pop_back();

        return res ;
    }
};

File: 0151-reverse-words-in-a-string/solution.java
class Solution {
    public String reverseWords(String s) {
        s+=" ";
        char[] c = s.toCharArray();
        int n = c.length;
        List<String> string_list = new ArrayList<>(); 
        String temp = "";
        for(int i = 0 ; i < n ; i++){
            if((c[i] != 32)){
                temp += c[i];
            }
            else if(temp.length() > 0){
                string_list.add(temp);
                temp = "";
            }
        }
        String res ="";
        int list_size = string_list.size();
        for(int i = list_size - 1 ; i >= 0 ; i--){
            if(string_list.get(i).length() > 0){
                res += string_list.get(i);
                if( i != 0)res += " ";
            }
        }
        return res ;
    }
}
